src/leap/mx/mail_receiver.py

```diff
@@ -378,10 +385,32 @@ def _step_process_mail_backend(self, filepath):
                 defer.returnValue(None)
 
             log.msg("Encrypting message to %s's pubkey" % (uuid,))
-            doc = yield self._encrypt_message(pubkey, mail_data)
+            try:
+                doc = yield self._encrypt_message(pubkey, mail_data)
+
+                yield self._export_message(uuid, doc)
+                yield self._remove(filepath)
+            except Exception as e:
+                yield self._bounce_with_timeout(filepath, msg, e)
+
+    @defer.inlineCallbacks
+    def _bounce_with_timeout(self, filepath, msg, error):
+        if filepath not in self._bounce_timeout:
+            self._bounce_timeout[filepath] = datetime.now()
+            log.msg("New staled email {0!r}: {1!r}".format(filepath, error))
+            defer.returnValue(None)
 
-            yield self._export_message(uuid, doc)
-            yield self._remove(filepath)
+        bounce_date = datetime.now() - timedelta(days=self.BOUNCE_TIMEOUT_DAYS)
+        if bounce_date > self._bounce_timeout[filepath]:
```

Comment:
somehow, I find that the comparison would be stated more clearly in terms of the delta:

MAX_BOUNCE_DELTA = timedelta(days=BOUNCE_TIMEOUT_DAYS)
current_delta = datetime.now() - bounce_start_timestamp
if current_delta > MAX_BOUNCE_DELTA:
  do_stuff ...